plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -3907,6 +3933,74 @@ private void testMultipleWriters(Session session, HiveStorageFormat storageForma
         }
     }
 
+    @Test
+    public void testMultipleWritersWhenTaskScaleWritersIsEnabled()
+    {
+        long workers = (long) computeScalar("SELECT count(*) FROM system.runtime.nodes");
+        int taskMaxScaleWriterCount = 4;
+        ImmutableList.of(PARQUET, ORC).forEach(format ->
+                assertFilesCount(getSession(), format, DataSize.of(200, KILOBYTE), taskMaxScaleWriterCount, false)
+                        .isBetween(workers + 1, workers * taskMaxScaleWriterCount));
+    }
+
+    @Test
+    public void testSingleWriterWhenTaskScaleWritersIsEnabled()
+    {
+        long workers = (long) computeScalar("SELECT count(*) FROM system.runtime.nodes");
+        // In the case of streaming, the number of writers is equal to the number of workers
+        testWithAllStorageFormats((session, storageFormat) ->
+                assertFilesCount(session, storageFormat, DataSize.of(2, GIGABYTE), 4, false)
+                        .isEqualTo(workers));
+    }
+
+    @Test
+    public void testWritersAcrossMultipleWorkersWhenScaleWritersIsEnabled()
+    {
+        long workers = (long) computeScalar("SELECT count(*) FROM system.runtime.nodes");
+        int taskMaxScaleWriterCount = 4;
+        // It is only applicable for pipeline execution mode, since we are testing
+        // when both "scaleWriters" and "taskScaleWriters" are enabled, the writers are
+        // scaling upto multiple worker nodes.
+        ImmutableList.of(PARQUET, ORC).forEach(format ->
+                assertFilesCount(getSession(), format, DataSize.of(200, KILOBYTE), taskMaxScaleWriterCount, true)
+                        .isBetween((long) taskMaxScaleWriterCount, workers * taskMaxScaleWriterCount));
+    }
+
+    protected AbstractLongAssert<?> assertFilesCount(
```

Comment:
rename `testTaskScaleWriters(...)`, `assertFilesCount` does not imply action inside